ui/org.eclipse.pde.core/src/org/eclipse/pde/core/plugin/IPluginReference.java

```diff
@@ -70,4 +70,15 @@ public interface IPluginReference extends IIdentifiable, IMatchRules {
 	 * @throws CoreException if the model is not editable
 	 */
 	void setVersion(String version) throws CoreException;
+
+	void setFilter(String filter) throws CoreException;
+
+	/**
+	 * Returns the filter string for this plugin reference.
+	 *
+	 * @return the filter string, or null if not set
+	 * @since 3.21
+	 */
+	String getFilter();
+
```

Comment:
Why did you add these methods this high in the type hierarchy? Wouldn't it be sufficient to add in `IPluginImport`?
Furthermore the set method is missing the `@since` tag and proper documentation. Then an API filter should not be necessary.